addons/mail/static/src/views/open_chat_hook.js

```diff
@@ -0,0 +1,27 @@
+/** @odoo-module **/
+
+import { useService } from "@web/core/utils/hooks";
+
+export const helpers = {
+    SUPPORTED_M2X_AVATAR_MODELS: ["res.users"],
+    buildOpenChatParams: (resModel, id) => {
+        if (resModel === "res.users") {
+            return { userId: id };
+        }
+    },
+}
```

Comment:
not a big big fan of this. I wonder if the maintenance of this wouldn't be simpler with an argument given directly to the useOpenChat. Here sure it's dead easy to use, but you have to patch it for all the possible extensions.